cmd/fs-v1.go

```diff
@@ -20,113 +20,239 @@ import (
 	"crypto/md5"
 	"crypto/sha256"
 	"encoding/hex"
+	"encoding/json"
 	"errors"
 	"fmt"
 	"hash"
 	"io"
-	"path"
+	"io/ioutil"
+	"os"
+	slashpath "path"
+	"path/filepath"
+	"runtime"
 	"sort"
 	"strings"
+	"syscall"
 
+	"github.com/minio/minio/pkg/disk"
+	"github.com/minio/minio/pkg/lock"
 	"github.com/minio/minio/pkg/mimedb"
 )
 
 // fsObjects - Implements fs object layer.
 type fsObjects struct {
-	storage StorageAPI
```

Comment:
I see we can't use existing posix.go and we short-circuit here.  I would think of an `shared-posix.go` having shared `FS` friendly layer.  Note that `shared-posix.go` != `StorageAPI`.  This will bring some clean abstraction.